src/test/java/com/iexec/core/task/update/TaskUpdateRequestManagerTests.java

```diff
@@ -91,22 +94,25 @@ void shouldNotUpdateAtTheSameTime() {
         final Random random = new Random();
         // Consuming a task update should only log the call a few times, while sleeping between each log
         // so that another task could be updated at the same time if authorized by lock.
-        final TaskUpdateRequestConsumer consumer = chainTaskId -> {
-            final int updateId = (int)System.nanoTime() % Integer.MAX_VALUE;
-            taskForUpdateId.put(updateId, chainTaskId);
-            for (int i = 0; i < callsPerUpdate; i++) {
-                try {
-                    TimeUnit.MILLISECONDS.sleep(random.nextInt(10));
-                } catch (InterruptedException ignored) {}
-                callsOrder.add(updateId);
-            }
-        };
         final ConcurrentMap<String, Object> locks = ExpiringMap.builder()
                 .expiration(100, TimeUnit.HOURS)
                 .build();
+        final Consumer<String> taskUpdater = chainTaskId -> {
+            synchronized (locks.computeIfAbsent(chainTaskId, key -> new Object())) {
+                final int updateId = (int) System.nanoTime() % Integer.MAX_VALUE;
+                taskForUpdateId.put(updateId, chainTaskId);
+                for (int i = 0; i < callsPerUpdate; i++) {
+                    try {
+                        TimeUnit.MILLISECONDS.sleep(random.nextInt(10));
+                    } catch (InterruptedException ignored) {
+                    }
```

Comment:
Isn't this dangerous, even if we are in tests ?

Reply:
I don't think so. This exception is only thrown when the thread is interrupted. That means the test won't complete, so it'll be considered as failed.
We could add some log if you think it is useful, I don't have any opinion on that.

Reply:
Maybe just to be safe and be sure the test will fail and stop fast enough
```
} catch (InterruptedException e) {
    throw new RuntimeException(e);
}
```
https://stackoverflow.com/questions/60619543/how-to-throw-an-exception-in-a-consumer-java-8

Reply:
Ok, let's do this!